src/busterbunny.js

```diff
@@ -70,14 +70,25 @@ module.exports = (function() {
         };
 
         self.raiseEvents = function (eventId, event, options, afterRaised) {
-            _stats.queuedEventsToRaise++;
-            if (options instanceof Function) {
+            if(options instanceof Function && afterRaised)
+                throw new Error('the third argument must be an object when a fourth argument is provided.');
+
+            if (options instanceof Function && !afterRaised) {
                 afterRaised = options;
                 options = null;
             }
 
-            var eventData = self.encoder.encodeEvent(event);
-            publishOrQueue(config.exchange, eventId, eventData, options, afterRaised);
+            if(!(afterRaised instanceof Function))
```

Comment:
What you have there is effectively what is here: https://github.com/airbnb/javascript#blocks.  I will adopt this approach.